crates/hir-def/src/item_tree.rs

```diff
@@ -267,14 +268,17 @@ impl ItemTree {
             macro_rules.shrink_to_fit();
             macro_defs.shrink_to_fit();
 
-            vis.arena.shrink_to_fit();
+            if let Some(set) = &mut vis.set {
+                set.shrink_to_fit();
+            }
         }
     }
 }
 
 #[derive(Default, Debug, Eq, PartialEq)]
 struct ItemVisibilities {
-    arena: Arena<RawVisibility>,
+    // `IndexSet`` is *big*, but we seldom have `in path` visibilities, so be lazy about it
+    set: Option<Box<IndexSet<RawVisibility>>>,
```

Comment:
Should use `FxIndexSet`.

Reply:
We probably need to add it to clippy's disallowed-types.

Reply:
> We probably need to add it to clippy's disallowed-types.

Right that only lits the std types